test/module/irohad/validation/chain_validation_test.cpp

```diff
@@ -15,105 +15,161 @@
  * limitations under the License.
  */
 
+#include "builders/protobuf/block.hpp"
 #include "module/irohad/ametsuchi/ametsuchi_mocks.hpp"
 #include "module/irohad/model/model_mocks.hpp"
 #include "validation/impl/chain_validator_impl.hpp"
 
+// TODO: 14-02-2018 Alexey Chernyshov remove after relocation to shared_model
+#include "backend/protobuf/from_old_model.hpp"
+
 using namespace iroha;
 using namespace iroha::model;
 using namespace iroha::validation;
 using namespace iroha::ametsuchi;
 
-using ::testing::_;
 using ::testing::A;
 using ::testing::ByRef;
 using ::testing::InvokeArgument;
 using ::testing::Return;
+using ::testing::_;
 
 class ChainValidationTest : public ::testing::Test {
  public:
   void SetUp() override {
     validator = std::make_shared<ChainValidatorImpl>();
     storage = std::make_shared<MockMutableStorage>();
     query = std::make_shared<MockWsvQuery>();
-
-    peer.pubkey.fill(2);
     peers = std::vector<Peer>{peer};
+  }
 
-    block.sigs.emplace_back();
-    block.sigs.back().pubkey = peer.pubkey;
-    block.prev_hash.fill(0);
-    hash = block.prev_hash;
+  /**
+   * Get block builder to build blocks for tests
+   * @return block builder
+   */
+  auto getBlockBuilder() const {
+    return shared_model::proto::BlockBuilder()
+        .transactions(std::vector<shared_model::proto::Transaction>{})
+        .txNumber(0)
+        .height(1)
+        .prevHash(hash)
+        .createdTime(iroha::time::now());
   }
 
+  // TODO: 14-02-2018 Alexey Chernyshov make sure peer has valid key after
+  // replacement with shared_model https://soramitsu.atlassian.net/browse/IR-903
   Peer peer;
   std::vector<Peer> peers;
-  Block block;
-  hash256_t hash;
+
+  shared_model::crypto::Hash hash =
+      shared_model::crypto::Hash(std::string(32, '0'));
```

Comment:
What is the point to use effective implementation of hashes?

Reply:
If we use another length of hash it will cause segfault in makeOldModel

Reply:
What's the problem?